Updates to upgrade procedures

kcarmichael08 opened the pull request:
Version(s):

- Merge to `rhacs-docs-main`
- Cherry pick to `rhacs-docs-4.3`, `rhacs-docs-4.4`, and `rhacs-docs-4.5` (but 4.5 will have further updates - see https://issues.redhat.com/browse/ROX-24162)

Issues:

- https://issues.redhat.com/browse/ROX-18594
- https://issues.redhat.com/browse/ROX-18976
- https://issues.redhat.com/browse/ROX-22204

Link to docs previews:

- [65388--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/installing_cloud_ocp/cloud-install-operator.html](https://65388--ocpdocs-pr.netlify.app/openshift-acs/latest/cloud_service/installing_cloud_ocp/cloud-install-operator.html)
- https://65388--ocpdocs-pr.netlify.app/openshift-acs/latest/upgrading/upgrade-helm.html
- https://65388--ocpdocs-pr.netlify.app/openshift-acs/latest/upgrading/upgrade-operator.html
- https://65388--ocpdocs-pr.netlify.app/openshift-acs/latest/upgrading/upgrade-roxctl.html

QE review: **ACS has no QE, approved by SMEs**
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:

Replaces #63482 and #63530

See also #65656

msugakov commented on upgrading/upgrade-operator.adoc:
```diff
@@ -9,15 +9,24 @@ toc::[]
 
-//If you installed {product-title-short} using the Operator and selected *Automatic* in the *Update approval* field, {product-title-short} is automatically updated when a new software version is released. If you selected *Manual*, you must approve subsequent Operator updates by using Operator Lifecycle Manager (OLM). For more information, see link:https://access.redhat.com/documentation/en-us/ope ...
+* For upgrading from version 3.74 to 4.x, see the following resources:
+** link:https://docs.openshift.com/acs/3.74/upgrading/upgrade-operator.html[Upgrading by using the Operator]
+** link:https://docs.openshift.com/acs/3.74/upgrading/upgrade-helm.html [https://docs.openshift.com/acs/3.74/upgrading/upgrade-helm.html]
+** link:https://docs.openshift.com/acs/3.74/upgrading/upgrade-roxctl.html[Manually Manually upgrading using the roxctl CLI]
```
1. Would having links to docs.openshift.com/acs/* constitute a problem for this documentation when it's rendered under access.redhat.com (i.e. <https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/4.2/html/upgrading/upgrade-operator>)?
It feels a bit odd that links will take users outside of access.redhat.com.
2. Something's a bit broken with the Helm link, see the screenshot.
![image](https://github.com/openshift/openshift-docs/assets/537715/b425a836-35e2-4899-922d-68c369f85e91)

kcarmichael08 replied:
Yeah, still fixing links and stuff. But good point - I'll change the links to the Customer Portal since that's the preferred way. (Just users accessing from docs.openshift.com will have the issue that they get taken to the Customer Portal.) Unfortunately there is no way to use relative links to a different version when you are viewing a specific version - i.e., if I use a relative link in 4.0, it will always go to the 4.0 version of the doc.

msugakov commented on upgrading/upgrade-operator.adoc:
```diff
@@ -9,15 +9,24 @@ toc::[]
 [role="_abstract"]
 Upgrades through the {rh-rhacs-first} Operator are performed automatically or manually, depending on the *Update approval* option you chose at installation.
```
I suppose the PR is WIP, although I just want to note that similar changes may be needed to `Upgrading using Helm charts` and `Upgrading using the roxctl CLI` pages so that they are consistent.

kcarmichael08 commented:
/reopen

ludydoo commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,7 +7,11 @@ include::modules/common-attributes.adoc[]
-You can upgrade to the latest version of {product-title} from a supported older version. For upgrading to {product-title-short} 4.0, you must be using the latest patch release of {product-title-short} 3.74. If you are using an older version, you must first upgrade to {product-title-short} 3.74.
+When upgrading, follow this guidance:
+
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
```
Should this be for both Central/SecuredCluster ? 
```suggestion
* If the installed version is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
```

kcarmichael08 replied:
Not sure, I assume so but I think @msugakov had specifically called out Central here, so maybe he can weigh in.

msugakov replied:
Only Central. This is because of rocksdb->postgres database migrations.

@ludydoo if you are aware of the similar need for the Secured Cluster, **this** is a perfect time to mention that!

ludydoo replied:
I am not aware of any such thing

msugakov replied:
I suppose we can close this thread as obsolete then.

ludydoo commented on upgrading/upgrade-operator.adoc:
```diff
@@ -9,15 +9,11 @@ toc::[]
-* Because of the database related changes introduced in {product-title-short} 4.0, even if you have selected *Automatic* in the *Update approval* field, you must manually upgrade to {product-title-short} 4.0.
-* You must be using {product-title-short} 3.74 to upgrade to {product-title-short} 4.0. If you are using a version older than 3.74, you must first upgrade to {product-title-short} 3.74 and then upgrade to {product-title-short} 4.0.
-====
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
```
Same question

msugakov replied:
```suggestion
* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central to version 3.74, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/upgrade-operator[upgrade documentation for version 3.74].
```

ludydoo commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,7 +7,11 @@ include::modules/common-attributes.adoc[]
+When upgrading, follow this guidance:
+
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
+* For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```
```suggestion
* For Operator- and Helm-based installations, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```

ludydoo commented on upgrading/upgrade-operator.adoc:
```diff
@@ -9,15 +9,11 @@ toc::[]
-* You must be using {product-title-short} 3.74 to upgrade to {product-title-short} 4.0. If you are using a version older than 3.74, you must first upgrade to {product-title-short} 3.74 and then upgrade to {product-title-short} 4.0.
-====
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
+* For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```
```suggestion
* For Operator- and Helm-based installations, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```

msugakov commented on modules/install-acs-operator-cloud.adoc:
```diff
@@ -29,7 +29,7 @@ Using the OperatorHub provided with {ocp} is the easiest way to install the {pro
 If you select automatic updates, when a new version of the Operator is available, Operator Lifecycle Manager (OLM) automatically upgrades the running instance of your Operator.
 +
-If you select manual updates, when a newer version of the Operator is available, OLM creates an update request. As a cluster administrator, you must manually approve the update request to update the Operator to the latest version.
+If you select manual updates, when a newer version of the Operator is available, OLM creates an update request. As a cluster administrator, you must manually approve the update request to update the Operator to the latest version. Red{&nbsp}Hat recommends maintaining secured cluster components at the same version as Central it is connected to for optimal compatibility. For {product-title-managed- ...
```
`{&nbsp}` does not render correctly.

![image](https://github.com/openshift/openshift-docs/assets/537715/c7e85c77-900d-46cf-afd8-ba4356b27d00)

----

I realized this page is dedicated to ACS CS therefore the only case to install ACS Operator is for Secured Cluster. There is no need to install ACS Central in ACS CS _by definition_.

Therefore, we should reword this paragraph with Secured Cluster being the only thing in the context. For Secured Cluster connected to ACS CS, we (or Red Hat) recommend automatic upgrades.

```suggestion
If you select manual updates, when a newer version of the Operator is available, OLM creates an update request. As a cluster administrator, you must manually approve the update request to update the Operator to the latest version.

Red{&nbsp}Hat recommends enabling automatic upgrades for Operator in {product-title-managed-short}. See the link:https://access.redhat.com/articles/7045053[Red Hat Advanced Cluster Security for Kubernetes Support Matrix] for more information.
```

msugakov commented on modules/prepare-operator-upgrades.adoc:
```diff
@@ -6,8 +6,14 @@
 * If you are upgrading from version 3.74, verify that you are running the latest patch release version of the {product-title-short} Operator 3.74.
 * Backup your existing Central database.
-* If the cluster you are upgrading contains the `SecuredCluster` custom resource (CR), change the collection method to `EBPF` or `CORE_BPF`.
+* If the `SecuredCluster` custom resource is present in the cluster, ensure the per node collection setting is set to EBPF by following these steps:
```
Hm. The suggestion to set to `EBPF` seems to come from the past. I believe, `CORE_BPF` is aimed to be the default from 4.5.0.

Let's ask the @erthalion @robbycochran @Stringy to double-check here.
Preview is https://65388--ocpdocs-pr.netlify.app/openshift-acs/latest/upgrading/upgrade-operator#prepare-operator-upgrades_upgrade-operator

JoukoVirtanen replied:
`CORE_BPF` is already the default in 4.4.

msugakov replied:
Thanks @JoukoVirtanen this is valuable info but it is still not enough for the question I asked. I asked if it is correct to ask users set collection method to EBPF on upgrading? We're talking about upgrade instructions to version 4.5.0.

erthalion replied:
How is the upgrade procedure special in this context?

I see following options:

* A regular one-version upgrade from 4.4 to 4.5 . As Jouko has mentioned, `CORE_BPF` is already a default one in 4.4, and we recommend to not stick to `EBPF` if not strictly needed.
* An upgrade from older version where `CORE_BPF` wasn't a default collection method. This instruction still doesn't make sense, because the last upgrade to 4.5 will render it useless.

I assume it comes from the previous deprecation of "kernel modules" collection method (might be the case some manual intervention was needed back then to switch to `EBPF`). Even more, the instructions described here wont be enough to enforce `EBPF`, as the operator contains translation logic that will still make the cluster use `CORE_BPF`.

To summarize, I don't think ensuring `EBPF` is needed. Does it make sense to you?

msugakov replied:
From what we discussed, both old content and new content appear to be invalid. Could you suggest something more correct, please?

I think we need to consider cases (remember we upgrade to 4.5.0 in this iteration)
1. Upgrading from pre-4.1.0 when KERNEL_MODULE was around.
2. Upgrading from [4.1.0, 4.4.0) when CORE_BPF was not available.
3. Upgrading from 4.4.0 or later.

In case of (1) the collection method MUST be switched to EBPF otherwise bad things happen but I think we wrote about it in docs as well, did not we? If yes, it seems the documentation becomes fragmented.

In cases (2) and (3), I understand it does not matter what collection method is because CORE_BPF will actually be used. In the case (3) it will be nice to tell to switch to CORE_BPF so that the config is consistent with the reality but this may confuse folks who want to force EBPF so I'm not sure if we should bother.

erthalion replied:
If it would be up to me, I would remove instructions around changing the collection method altogether, and instead add a warning at the end of this page saying "if you upgrade from version 4.1.0 or below, make sure to the collection method is CORE_BPF" and explanation how to verify this and modify if needed. How does it soun?

kcarmichael08 replied:
I think that this is obsolete after Jouko and I made changes regarding the collection method during the last release. See:

https://docs.openshift.com/acs/4.4/upgrading/upgrade-operator.html#change-collection-method_upgrade-operator

I'll change the language and add a link (cannot add a link directly in the text, but will add it under additional references - this is a limitation of our publishing system).

kcarmichael08 replied:
Actually, the Change Collection Method section is directly after this "preparing" section, so...I don't think it needs a link in additional references.

msugakov commented on modules/remove-central-attached-pv-overview.adoc:
```diff
@@ -3,7 +3,7 @@
 :_mod-docs-content-type: CONCEPT
 [id="remove-central-attached-pv-overview_{context}"]
-= Remove Central-attached PV
+= Remove Central-attached PV (version 4.1 and later)
```
When I look at the preview of this section <https://65388--ocpdocs-pr.netlify.app/openshift-acs/latest/upgrading/upgrade-operator#remove-central-attached-pv-overview_upgrade-operator>, it troubles me a bit that we don't mention the removal should be done _after_ the operator has upgraded the product. Also, `(version 4.1 and later)` may read ambiguous (whether it is _from_ version 4.1 and newer or _to_ version 4.1 and newer). It could be less ambiguous although lengthier to say `(after upgrading to version 4.1 and later)`.

dashrews78 replied:
Couple lines down:

"After completing the upgrade to 4.1 you can remove the Central-attached persistent volume claim (PVC) to free up the storage if you do not plan to rollback or restore from earlier RocksDB backups."

msugakov commented on modules/updates-and-upgrades.adoc:
```diff
@@ -12,4 +12,5 @@ Customers have no control over when a Central service update occurs. For more in
-Customers are responsible for Secured Cluster services upgrades required to maintain this version compatibility.
\ No newline at end of file
+Customers are responsible for Secured Cluster services upgrades required to maintain this version compatibility.
+Red{&nbsp}Hat recommends maintaining secured cluster components at the same version as the Central it is connected to for optimal compatibility. The software version for Central is automatically updated. See the link:https://access.redhat.com/articles/7045053[Red Hat Advanced Cluster Security for Kubernetes Support Matrix] for more information.
```
Since this module is part of CS only:
- There is **no** such thing as a **special Secured Cluster for CS**. It's always the same Secured Cluster _distribution_ for both CS and self-managed (the difference is _where_ one _connects to_: self-managed central or CS central).
- Only customers are responsible for Secured Cluster upgrades.
- At the same time we strongly recommend always upgrading Secured Cluster to the latest version shortly after the one gets released.
- Therefore we strongly recommend enabling automatic upgrades.

If I try to write it down, it'd look something like this instead of the current content.
```suggestion
Customers are responsible for timely {product-title-short} Secured Cluster services upgrades required to maintain compatibility with {product-title-managed-short}.
Red Hat recommends enabling automatic upgrades for Secured Clusters connected to  {product-title-managed-short}.
See the link:https://access.redhat.com/articles/7045053[Red Hat Advanced Cluster Security for Kubernetes Support Matrix] for more information.
```

msugakov commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,7 +7,11 @@ include::modules/common-attributes.adoc[]
-You can upgrade to the latest version of {product-title} from a supported older version. For upgrading to {product-title-short} 4.0, you must be using the latest patch release of {product-title-short} 3.74. If you are using an older version, you must first upgrade to {product-title-short} 3.74.
+When upgrading, follow this guidance:
+
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
```
This looks plain wrong to me. Right in the **next bullet point** we tell about upgrading from 3.74 to 4.0-4.4, we don't need to tell the same thing here ... which we actually don't because the provided link is correct.

```suggestion
* If Central is earlier than 3.74, you must upgrade to the latest 3.74 patch before upgrading to a 4.x release. See the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74] regarding upgrades from earlier versions to 3.74.
```

msugakov reviewed (commented):
Partial review. Need to allocate more time to complete going through this change.

msugakov commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,7 +7,11 @@ include::modules/common-attributes.adoc[]
+When upgrading, follow this guidance:
+
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
+* For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```
1. We don't need to mention Operator here because these are instructions for Helm.
2. I suggest to make it more explicit that 3.74 -> 4.0..4.4 is ok but 3.74 -> 4.5+ is not ok.
3. I like your point about upgrading to the latest but I suggest to reword it slightly so that the proposed action is clear.

```suggestion
* When upgrading Helm-based Central deployments from release 3.74, choose any latest patch of {product-title-short} version 4.0 through 4.4 but not 4.5 or later. After that, upgrade Central to the latest version.
```

Also, a link doc from 4.4's documentation would be useful.

kcarmichael08 replied:
> 2\. I suggest to make it more explicit that 3.74 -> 4.0..4.4 is ok but 3.74 -> 4.5+ is not ok.

This PR isn't for 4.5, this is for 4.4 and earlier. I'll make a separate PR for 4.5.

kcarmichael08 replied:
Rewrote because the intent was to tell them to upgrade to 4.4.x (the latest version, as of this publication - 4.4 version).

kcarmichael08 replied:
And I can add a link to the 4.4 docs from the 4.5 docs if you think that's necessary. (per your comment above)

msugakov commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,7 +7,11 @@ include::modules/common-attributes.adoc[]
+
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
+* For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
+* For upgrading within version 4.x, see the latest version of the documentation.
```
```suggestion
```
Because this page _is_ the latest version of the documentation.

msugakov commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,7 +7,11 @@ include::modules/common-attributes.adoc[]
 
 [role="_abstract"]
-You can upgrade to the latest version of {product-title} from a supported older version. For upgrading to {product-title-short} 4.0, you must be using the latest patch release of {product-title-short} 3.74. If you are using an older version, you must first upgrade to {product-title-short} 3.74.
+When upgrading, follow this guidance:
```
I'm nitpicking but this sentence looks very bare bones to me. How about if we make it a sub-section and call it like "upgrade sequence from ACS 3.x"?

kcarmichael08 replied:
This is the abstract, which is a specific field that the Customer Portal requires to provide an overview of what is discussed in the module. I'll add something else for the abstract.

msugakov commented on upgrading/upgrade-operator.adoc:
```diff
@@ -9,15 +9,11 @@ toc::[]
-* You must be using {product-title-short} 3.74 to upgrade to {product-title-short} 4.0. If you are using a version older than 3.74, you must first upgrade to {product-title-short} 3.74 and then upgrade to {product-title-short} 4.0.
-====
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
+* For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```
Same suggestion as I did for Helm.

dashrews78 replied:
An alternative wording.  Basically the same

* When upgrading Helm-based Central deployments from release 3.74, first ensure Operator upgrade mode is set to Manual, upgrade the Operator to 4.0 following the procedure in the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/4.0/html/upgrading/upgrade-operator[upgrade documentation for version 4.0], and ensure Central comes up online. Once the upgrade to 4.0 is complete, Red Hat recommends upgrading central to the latest version for full functionality.

msugakov commented on upgrading/upgrade-operator.adoc:
```diff
@@ -9,15 +9,11 @@ toc::[]
-====
+* If Central is earlier than 3.74, you must upgrade to 3.74 before upgrading to a 4.x release. For upgrading Central from version 3.74 to 4.x, see the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74].
+* For Operator- and Helm-based Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
+* For upgrading within version 4.x, see the latest version of the documentation.
```
```suggestion
```
Same as with Helm.

msugakov reviewed (commented):
Another partial review.

msugakov commented on upgrading/upgrade-roxctl.adoc:
```diff
@@ -12,7 +12,8 @@ You can upgrade to the latest version of {rh-rhacs-first} from a supported older
 ====
 * You need to perform the manual upgrade procedure only if you used the `roxctl` CLI to install {product-title-short}.
-* For upgrading to {product-title-short} 4.0, you must be using the latest patch release of {product-title-short} 3.74. If you are using an older version, you must first upgrade to {product-title-short} 3.74 before upgrading to {product-title-short} 4.0.
+* For manifest-based (roxctl) Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to the latest version.
```
```suggestion
* For manifest-based (roxctl) Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4 but not 4.5 or later. After that, upgrade Central to the latest version.
```

dashrews78 replied:
1st bullet says you can do directly to latest.  2nd bullet recommends you don't.  Should we flip the order on those?

Also a suggestion on the first bullet:

* For manifest-based (roxctl) Central deployments, you can upgrade from release 3.74 to any latest patch of {product-title-short} version 4.0 through 4.4.  It is not possible to upgrade from 3.74 directly to 4.5 or later.  For full functionality Red Hat recommends upgrading central to 4.0 through 4.4 and then to the latest version

msugakov commented:
I think, after addressing these points we'll just get the first approximation of what's required to to make the upgrade instructions straightforward for users.

There's a lot more to the structure that I see needs a rework although I'm afraid to pull this in because this PR already has a history from the August 2023.
My suggestion would be to deal with this PR in the current scope and do a pair programming session (over Google Meet) on the upgrade instructions so that we can interactively define a direction of improvement.

kcarmichael08 commented:
> I think, after addressing these points we'll just get the first approximation of what's required to to make the upgrade instructions straightforward for users.
> 
> There's a lot more to the structure that I see needs a rework although I'm afraid to pull this in because this PR already has a prior history from the August 2023. My suggestion would be to deal with this PR in the current scope and do a pair programming session (over Google Meet) on the upgrade instructions so that we can interactively define a direction of improvement.

IMO after struggling with this PR and its many comments previously, and then trying to take it back up again so that it can be closed out, it's difficult to make the instructions straightforward because the *process itself* isn't straightforward. I'm happy to work on another iteration after this PR is done, but I also have to start work on 4.5 documentation. Maybe after I push the changes after your latest comments, we can do a google meet session and go over the doc line by line. Or, alternatively, you can pull the PR and propose changes yourself, if you think that would work better.

kcarmichael08 commented:
I think I have everyone's suggestions incorporated, except for a 4.5 version (which I will wait to create until after we get this one completed).

msugakov reviewed (approved):
Hopefully the last round of comments from me. I hope my suggestions don't contradict the earlier ones, otherwise please let me know :smile_cat:. Approving as I think it's mergeable after addressing these.

kcarmichael08 commented on modules/install-acs-operator-cloud.adoc:
```diff
@@ -30,6 +30,8 @@ Using the OperatorHub provided with {ocp} is the easiest way to install the {pro
 +
 If you select manual updates, when a newer version of the Operator is available, OLM creates an update request. As a cluster administrator, you must manually approve the update request to update the Operator to the latest version.
++
+Red{&nbsp}Hat recommends enabling automatic upgrades for Operator in {product-title-managed-short}. See the link:https://access.redhat.com/articles/7045053[Red Hat Advanced Cluster Security for Kubernetes Support Matrix] for more information.
```
Argh, thanks for catching this.

kcarmichael08 commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,11 +7,19 @@ include::modules/common-attributes.adoc[]
+When upgrading from earlier releases, follow this guidance:
+
+* If the release for Central is earlier than 3.74, you must upgrade to the latest 3.74 patch before upgrading to a 4.x release. See the link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.74/html/upgrading/index[upgrade documentation for version 3.74] for information about upgrades from earlier versions to 3.74.
+* When upgrading Helm-based installations from release 3.74, you can upgrade to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to release 4.4.
```
Created https://issues.redhat.com/browse/ROX-24162 for 4.5+

kcarmichael08 commented on upgrading/upgrade-roxctl.adoc:
```diff
@@ -12,7 +12,8 @@ You can upgrade to the latest version of {rh-rhacs-first} from a supported older
 * You need to perform the manual upgrade procedure only if you used the `roxctl` CLI to install {product-title-short}.
-* For upgrading to {product-title-short} 4.0, you must be using the latest patch release of {product-title-short} 3.74. If you are using an older version, you must first upgrade to {product-title-short} 3.74 before upgrading to {product-title-short} 4.0.
+* There are manual steps for each version upgrade that must be followed, for example, from version 3.74 to version 4.0, and from version 4.0 to version 4.1. Therefore, Red{nbsp}Hat recommends upgrading first from 3.74 to 4.0, then from 4.0 to 4.1, then 4.1 to 4.2, until the selected version is installed.
+* For manifest-based (roxctl) Central deployments, you can upgrade to any latest patch of {product-title-short} version 4.0 through 4.4. However, for full functionality, upgrade to version 4.4.
```
I think originally it was meant to advise people to always upgrade to the latest version to get the latest features, but I think that's implicit anyway, so I'll delete.

kcarmichael08 replied:
took your suggestion in the next comment - so this is fixed

maxwelldb commented on modules/prepare-operator-upgrades.adoc:
```diff
@@ -6,8 +6,8 @@
 * If you are upgrading from version 3.74, verify that you are running the latest patch release version of the {product-title-short} Operator 3.74.
 * Backup your existing Central database.
-* If the cluster you are upgrading contains the `SecuredCluster` custom resource (CR), change the collection method to `EBPF` or `CORE_BPF`.
+* If the cluster you are upgrading contains the `SecuredCluster` custom resource (CR), change the collection method to EBPF or CORE_BPF. See "Changing the collection method" in the next section.
```
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#lead-in-link-sentence-style-for-non-xref-situations

maxwelldb commented on modules/remove-central-attached-pv-operator.adoc:
```diff
@@ -3,7 +3,7 @@
 :_mod-docs-content-type: PROCEDURE
 [id="remove-central-attached-pv-operator_{context}"]
-= Remove Central-attached PV using the {product-title-short} Operator
+= Remove Central-attached PV using the {product-title-short} Operator (version 4.1 and later)
```
Ideally, this would be rewritten to exclude the parens. I don't think this is a total prohibition on it, but ISG generally allows them only as a last resort.

kcarmichael08 replied:
It makes for a long heading but I rewrote.

maxwelldb commented on modules/remove-central-attached-pv-operator.adoc:
```diff
@@ -3,7 +3,7 @@
 :_mod-docs-content-type: PROCEDURE
 [id="remove-central-attached-pv-operator_{context}"]
-= Remove Central-attached PV using the {product-title-short} Operator
+= Remove Central-attached PV using the {product-title-short} Operator (version 4.1 and later)
```
Procedure titles should be gerund phrases per mod docs guidance.

kcarmichael08 replied:
That was existing content, which I was generally trying to avoid restructuring. I was just asked to specify the version info.

maxwelldb commented on modules/remove-central-attached-pv-overview.adoc:
```diff
@@ -3,14 +3,14 @@
 :_mod-docs-content-type: CONCEPT
 [id="remove-central-attached-pv-overview_{context}"]
-= Remove Central-attached PV
+= Remove Central-attached PV after upgrading to version 4.1 and later
```
Gerund phrase.

maxwelldb commented on modules/remove-central-attached-pv-overview.adoc:
```diff
@@ -3,14 +3,14 @@
 The PV has data and persistent files used by earlier {product-title-short} versions. You can use the PV to roll back to an earlier version before {product-title-short} 4.1. Or, if you have a large RocksDB backup bundle for Central, you can use the PV to restore that data.
 
-If you do not plan to roll back or restore from earlier RocksDB backups, you can remove the Central-attached persistent volume claim (PVC) to free up the storage.
+After completing the upgrade to 4.1, you can remove the Central-attached persistent volume claim (PVC) to free up the storage if you do not plan to roll back or restore from earlier RocksDB backups.
```
Thinking of something like this to make the language more active and simplify the first verb phrase: "After you upgrade RHACS to 4.1, you can remove the Central-attached persistent volume..."

maxwelldb replied:
Might consider breaking up the sentence. It's quite long.

maxwelldb commented on modules/updates-and-upgrades.adoc:
```diff
@@ -11,5 +11,8 @@ The decision regarding the need for a Service update to the Central instance and
-The version of {product-title-managed-short} used on Secured Clusters must match the version of the Central instance of {product-title-managed-short} to ensure compatibility.
-Customers are responsible for Secured Cluster services upgrades required to maintain this version compatibility.
\ No newline at end of file
+Customers are responsible for timely {product-title-short} Secured Cluster services upgrades required to maintain compatibility with {product-title-managed-short}.
```
```suggestion
Customers are responsible for timely {product-title-short} Secured Cluster services upgrades that are required to maintain compatibility with {product-title-managed-short}.
```
ISG specifies "that" for restrictive clauses.

maxwelldb commented on modules/updates-and-upgrades.adoc:
```diff
@@ -11,5 +11,8 @@ The decision regarding the need for a Service update to the Central instance and
\ No newline at end of file
+Customers are responsible for timely {product-title-short} Secured Cluster services upgrades required to maintain compatibility with {product-title-managed-short}.
+
+Red Hat recommends enabling automatic upgrades for Secured Clusters connected to  {product-title-managed-short}.
```
```suggestion
Red Hat recommends enabling automatic upgrades for Secured Clusters that are connected to {product-title-managed-short}.
```
or
```suggestion
Red Hat recommends enabling automatic upgrades for Secured Clusters that connect to {product-title-managed-short}.
```

kcarmichael08 replied:
I realize this is a weird distinction, but the clusters exist outside of RHACS and while they do connect to them, the component that manages the connection as a whole (Central) is within ACS so I think it's more clear to say "that are connected to", if that makes sense.

maxwelldb commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,15 +7,23 @@ include::modules/common-attributes.adoc[]
 
 [role="_abstract"]
-You can upgrade to the latest version of {product-title} from a supported older version. For upgrading to {product-title-short} 4.0, you must be using the latest patch release of {product-title-short} 3.74. If you are using an older version, you must first upgrade to {product-title-short} 3.74.
+Upgrading {product-title-short} requires a specific upgrade path depending on the release of {product-title-short} that you are running. You must also back up your Central database before updating the Helm chart and peforming the upgrade.
```
Could make this a bit more active. E.g., "You must follow a specific path to upgrade {product-title-short}..."

maxwelldb commented on upgrading/upgrade-helm.adoc:
```diff
@@ -7,15 +7,23 @@ include::modules/common-attributes.adoc[]
+If you have installed {product-title-short} by using Helm charts, to upgrade to the latest version of {product-title-short} perform the following steps:
+
+. Back up the Central database.
+. (Optional) Optimize Central database and Persistent Volume Claims (PVC).
+. (Optional) Generate `values-private.yaml` configuration file containing root certificates for the central-services Helm chart.
```
```suggestion
. Optional: Optimize central database and Persistent Volume Claims (PVC).
. Optional: Generate `values-private.yaml` configuration file containing root certificates for the central-services Helm chart.
```
or, because it's not a procedure, maybe: 
```suggestion
. Optionally, optimize central database and Persistent Volume Claims (PVC).
. Optionally, generate `values-private.yaml` configuration file containing root certificates for the central-services Helm chart.
```

kcarmichael08 replied:
Central is a component and is always capitalized in the RHACS docs.

maxwelldb reviewed (commented):
Made some comments and suggestions. Re-label when ready! 👍

kcarmichael08 commented:
/cherrypick rhacs-docs-4.4

kcarmichael08 commented:
/cherrypick rhacs-docs-4.5
